runtime/server/generate_metrics_view.go

```diff
@@ -0,0 +1,418 @@
+	}
+
+	return &runtimev1.GenerateMetricsViewFileResponse{AiSucceeded: aiSucceeded}, nil
+}
+
+// generateMetricsViewYAMLWithAI attempts to generate a metrics view YAML definition from a table schema using AI.
+// It validates that the result is a valid metrics view. Due to the unpredictable nature of AI (and chance of downtime), this function may error non-deterministically.
+func (s *Server) generateMetricsViewYAMLWithAI(ctx context.Context, instanceID, dialect, connector, tblName string, isDefaultConnector, isModel bool, schema *runtimev1.StructType) (string, error) {
+	// Build messages
+	msgs := []*drivers.CompletionMessage{
+		{Role: "system", Data: metricsViewYAMLSystemPrompt()},
+		{Role: "user", Data: metricsViewYAMLUserPrompt(dialect, tblName, schema)},
+	}
+
+	// Connect to the AI service configured for the instance
+	ai, release, err := s.runtime.AI(ctx, instanceID)
+	if err != nil {
+		return "", err
+	}
+	defer release()
+
+	// Apply timeout
+	ctx, cancel := context.WithTimeout(ctx, aiGenerateTimeout)
+	defer cancel()
+
+	// Call AI service to infer a metrics view YAML
+	res, err := ai.Complete(ctx, msgs)
+	if err != nil {
+		return "", err
+	}
+
+	// The AI may produce Markdown output. Remove the code tags around the YAML.
+	res.Data = strings.TrimPrefix(res.Data, "```yaml")
+	res.Data = strings.TrimPrefix(res.Data, "```")
+	res.Data = strings.TrimSuffix(res.Data, "```")
+
+	// Parse the YAML structure
+	var doc metricsViewYAML
+	if err := yaml.Unmarshal([]byte(res.Data), &doc); err != nil {
+		return "", fmt.Errorf("invalid metrics view YAML: %w", err)
+	}
+
+	// The AI only generates metrics. We fill in the other properties using the simple logic.
+	doc.TimeDimension = generateMetricsViewYAMLSimpleTimeDimension(schema)
+	doc.Dimensions = generateMetricsViewYAMLSimpleDimensions(schema)
+	if isModel {
+		doc.Model = tblName
+	} else {
+		if !isDefaultConnector {
+			doc.Connector = connector
+		}
+		doc.Table = tblName
+	}
+
+	// Validate the generated measures (not validating other parts since those are not AI-generated)
+	spec := &runtimev1.MetricsViewSpec{
+		Connector: connector,
+		Table:     tblName,
+	}
+	for _, measure := range doc.Measures {
+		spec.Measures = append(spec.Measures, &runtimev1.MetricsViewSpec_MeasureV2{
+			Name:         measure.Name,
+			Label:        measure.Label,
+			Expression:   measure.Expression,
+			FormatPreset: measure.FormatPreset,
+		})
+	}
+	validateResult, err := s.runtime.ValidateMetricsView(ctx, instanceID, spec)
+	if err != nil {
+		return "", err
+	}
+
+	// Remove all invalid measures. We do this in two steps to preserve the indexes returned in MeasureErrs:
+	// First, we set the invalid measures to nil. Second, we remove the nil entries.
+	for _, ie := range validateResult.MeasureErrs {
+		doc.Measures[ie.Idx] = nil
+	}
+	for idx := 0; idx < len(doc.Measures); {
+		if doc.Measures[idx] == nil {
+			doc.Measures = slices.Delete(doc.Measures, idx, idx+1)
+		} else {
+			idx++
+		}
+	}
+
```

Comment:
Emit a metric - 
{
"completiton_id" : "UUID", 
valid_metrics_count: 5, 
invalid_metrics_count: 4, 
user_id and other dimensions that we emit today..
}